pkgs/build-support/docker/default.nix

```diff
@@ -76,6 +78,30 @@ let
   # mapping from the go package.
   defaultArchitecture = go.GOARCH;
 
+  compressors = {
+    none = {
+      ext = "";
+      nativeInputs = [ ];
+      compress = "cat";
+      decompress = "cat";
+    };
+    gz = {
+      ext = ".gz";
+      nativeInputs = [ pigz ];
+      compress = "pigz -p$NIX_BUILD_CORES -nTR";
+      decompress = "pigz -d -p$NIX_BUILD_CORES";
```

Comment:
Drive-by fix-up: I also restricted pigz to the `NIX_BUILD_CORES` instead of letting it guess while I moved the command up here. Happy to put this into a separate commit if so desired.